### ISSUE TITLE
Issues with onChangeDate

### ISSUE THREAD

**Issue by rohanvasishth** [acro5piano/react-native-big-calendar#712]
I am having issues with the onChangeDate method prop.

I have a custom calendar strip (functional component) working off of a selectedDate state variable. However, anytime I change the state of selectedDate from onChangeDate, I get the following error:

Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

Here is my code:

```tsx
import React, { useCallback, useState } from 'react';
import { View, StyleSheet, Modal, Dimensions, Text, TouchableOpacity } from 'react-native';

import { Colors, Typography, Spacing } from '../../styles';
import { Label, FloatingActionButton } from '../../components/atoms';

import { CalendarStrip } from '../../components/molecules';
import { Calendar } from 'react-native-big-calendar'

import { isEqual } from 'date-fns';

import NewEvent from '../NewEvent';

const events = [...]

const deviceHeight = Dimensions.get('window').height
const calendarStripHeight = 30 + 30 + 50 + (2 * Spacing.S)

const Schedule = ({ navigation }: { navigation: any }) => {
    const [modalVisible, setModalVisible] = useState(false);
    const [selectedDate, changeSelectedDate] = useState(new Date())

    const calendarHeight = deviceHeight - Spacing.XXXL - calendarStripHeight - Spacing.XL - Spacing.L - (2 * Spacing.S)

    const handleNewCarpoolEventTapped = () => {
        setModalVisible(!modalVisible)
    };

    const onChangeDate = useCallback(([start, end]) => {
        changeSelectedDate(end)
    }, [])

    const renderEvent = <T extends ICalendarEventBase>(
        event: T,
        touchableOpacityProps: CalendarTouchableOpacityProps,
    ) => (
        <TouchableOpacity style={styles.eventContainer} {...touchableOpacityProps}>
            <Text style={styles.eventTitle}>{`${event.title}`}</Text>
            <Text style={styles.eventDescription}>{`${event.driver} | ${event.rider_count}`}</Text>
            <Text style={styles.eventLocation}>{`${event.location}`}</Text>
        </TouchableOpacity>
    )

    return (
        <View style={styles.container}>
            <Label
                text={'Schedule'}
                font={Typography.BOLD}
                size={Typography.L}
                color={Colors.PRIMARY}
                topSpacing={Spacing.XXXL}
                bottomSpacing={0}
            />
            <CalendarStrip
                selectedDate={selectedDate}
                markedDate={['2022-07-13', '2022-07-16']}
                onPressDate={(date: Date) => {
                    changeSelectedDate(date);
                }}
                onPressGoToday={(today: Date) => {
                    changeSelectedDate(today)
                }}
            />
            <Calendar
                events={events}
                height={calendarHeight}
                ampm={true}
                mode={'day'}
                date={selectedDate}
                renderHeader={() => null}
                scrollOffsetMinutes={600}
                swipeEnabled={true}
                eventCellStyle={styles.eventContainer}
                renderEvent={renderEvent}
                onChangeDate={onChangeDate}
                onPressEvent={(event) => console.log(event)}
            />
            <FloatingActionButton
                onPress={() => handleNewCarpoolEventTapped()}
            />
            <Modal
                animationType='slide'
                transparent={true}
                visible={modalVisible}
                onRequestClose={() => {
                    setModalVisible(!modalVisible);
                }}
                statusBarTranslucent
            >
                <NewEvent close={() => setModalVisible(!modalVisible)} />
            </Modal>
        </View>
    );
};
```

**Comment by acro5piano** [comment]
Hi, can you try to wrap your handler with `useCallbaek` ?

**Comment by rohanvasishth** [comment]
That doesn't fix the issue. here is the new code:

const Schedule = ({ navigation }: { navigation: any }) => {
    const [modalVisible, setModalVisible] = useState(false);
    const [selectedDate, changeSelectedDate] = useState(new Date())

    const calendarHeight = deviceHeight - Spacing.XXXL - calendarStripHeight - Spacing.XL - Spacing.L - (2 * Spacing.S)

    const handleNewCarpoolEventTapped = () => {
        setModalVisible(!modalVisible)
    };

    const onChangeDate = useCallback(([start, end]) => {
        changeSelectedDate(end)
    }, [selectedDate])

    const onPressEvent = useCallback((event) => {
        console.log(event)
    }, [])

    const renderEvent = <T extends ICalendarEventBase>(
        event: T,
        touchableOpacityProps: CalendarTouchableOpacityProps,
    ) => (
        <TouchableOpacity style={styles.eventContainer} {...touchableOpacityProps}>
            <Text style={styles.eventTitle}>{`${event.title}`}</Text>
            <Text style={styles.eventDescription}>{`${event.driver} | ${event.rider_count}`}</Text>
            <Text style={styles.eventLocation}>{`${event.location}`}</Text>
        </TouchableOpacity>
    )

    return (
        <View style={styles.container}>
            <Label
                text={'Schedule'}
                font={Typography.BOLD}
                size={Typography.L}
                color={Colors.PRIMARY}
                topSpacing={Spacing.XXXL}
                bottomSpacing={0}
            />
            <CalendarStrip
                selectedDate={selectedDate}
                markedDate={['2022-07-13', '2022-07-16']}
                onPressDate={(date: Date) => {
                    changeSelectedDate(date);
                }}
                onPressGoToday={(today: Date) => {
                    changeSelectedDate(today)
                }}
            />
            <Calendar
                events={events}
                height={calendarHeight}
                ampm={true}
                mode={'day'}
                date={selectedDate}
                renderHeader={() => null}
                scrollOffsetMinutes={(new Date()).getHours() * 60}
                swipeEnabled={true}
                eventCellStyle={styles.eventContainer}
                renderEvent={renderEvent}
                onChangeDate={onChangeDate}
                onPressEvent={onPressEvent}
            />
            <FloatingActionButton
                onPress={() => handleNewCarpoolEventTapped()}
            />
            <Modal
                animationType='slide'
                transparent={true}
                visible={modalVisible}
                onRequestClose={() => {
                    setModalVisible(!modalVisible);
                }}
                statusBarTranslucent
            >
                <NewEvent close={() => setModalVisible(!modalVisible)} />
            </Modal>
        </View>
    );
};

**Comment by rohanvasishth** [comment]
any other advice @acro5piano ?

**Comment by acro5piano** [comment]
@rohanvasishth 
I'm not sure what is the root cause. You can try to wrap `renderEvent` with `useCallback`. If that doesn't help, try to remove props one-by-one and detect which prop affecting the issue.

**Comment by TheTushar2696** [comment]
@acro5piano  The same issue is happening with me as well, function passed to onChangeDate runs infinitely, and goes in a loop.

**Comment by mubashiralisiddiqui** [comment]
> @rohanvasishth I'm not sure what is the root cause. You can try to wrap `renderEvent` with `useCallback`. If that doesn't help, try to remove props one-by-one and detect which prop affecting the issue.
as per my understanding
this is happen because of an extra useEffect at line num 144 and the extra dependency in another useEffect which is at line num 183 of CalenderContainer.tsx I have tested it by fixing that on my side and it's working fine
I can create MR for that but I am not sure for what reason you have done that can you recheck that, please
<img width="1053" alt="image" src="https://user-images.githubusercontent.com/16689214/196030472-72abccf2-1bf9-4cfb-a758-a0dfc89026b1.png">

**Comment by mubashiralisiddiqui** [comment]
@acro5piano 
https://github.com/acro5piano/react-native-big-calendar/pull/764
here is the MR for this issue  
this fix will resolve the issue but we are getting eslint error 
and I tried to fix it as per the recommended way which you also implement but it causing infinite setState
so I have to disable the es-lint  rule for this since its a major issue we can disable it
although I haven't push that change of eslint in this MR

**Comment by acro5piano** [comment]
@mubashiralisiddiqui Thanks. Let's continue to discuss it on the pull request page.

**Comment by acro5piano** [comment]
The fix was released on v3.3.1. Thank you very much @mubashiralisiddiqui !

I close this issue for now. If the problem continues, feel free to reopen this issue.

**Comment by acro5piano** [comment]
This issue was re-resolved on #801 , because #764 created a issue on swiping right to left.
